src/superqt/switch/_toggle_switch.py

```diff
@@ -179,109 +154,122 @@ def drawHandle(
         self.update()
-        return None
 
     onColor = Property(QtGui.QColor, _get_onColor, _set_onColor)
 
     def _get_offColor(self) -> QtGui.QColor:
-        return self._style_option.off_color
+        return self._off_color
 
     def _set_offColor(self, color: QtGui.QColor | QtGui.QBrush) -> None:
-        self._style_option.off_color = QtGui.QColor(color)
+        self._off_color = QtGui.QColor(color)
         self.update()
-        return None
 
     offColor = Property(QtGui.QColor, _get_offColor, _set_offColor)
 
     def _get_handleColor(self) -> QtGui.QColor:
-        return self._style_option.handle_color
+        return self._handle_color
 
     def _set_handleColor(self, color: QtGui.QColor | QtGui.QBrush) -> None:
-        self._style_option.handle_color = QtGui.QColor(color)
+        self._handle_color = QtGui.QColor(color)
         self.update()
-        return None
 
     handleColor = Property(QtGui.QColor, _get_handleColor, _set_handleColor)
 
     ### Other private methods ###
 
-    def _set_checked_animated(self, val: bool):
-        start = self._offset_for_checkstate(self._checked)
+    def _animate_handle(self, val: bool) -> None:
         end = self._offset_for_checkstate(val)
-
-        self._checked = val
-        self._anim.setStartValue(start)
-        self._anim.setEndValue(end)
-        self._anim.setDuration(120)
-        self._anim.start()
-        return None
+        if self._anim.duration():
+            self._anim.setStartValue(self._offset_for_checkstate(not val))
+            self._anim.setEndValue(end)
+            self._anim.start()
+        else:
+            self._set_offset(end)
 
     def _get_offset(self) -> float:
         return self._offset_value
 
     def _set_offset(self, offset: float) -> None:
         self._offset_value = offset
         self.update()
-        return None
 
     _offset = Property(float, _get_offset, _set_offset)
 
-    def _offset_for_checkstate(self, val: bool) -> int:
-        opt = self._style_option
+    def _offset_for_checkstate(self, val: bool) -> float:
+        opt = QStyleOptionToggleSwitch()
+        self.initStyleOption(opt)
         if val:
             offset = opt.margin + opt.switch_width - opt.switch_height / 2
         else:
             offset = opt.margin + opt.switch_height / 2
         return offset
 
-    def _prep_draw_groove(
-        self, p: QtGui.QPainter, opt: QStyleOptionToggleSwitch
-    ) -> QtCore.QRect:
-        p.setPen(Qt.PenStyle.NoPen)
-
-        # draw the groove
-        if self.isEnabled():
-            p.setBrush(opt.on_color if self.isChecked() else opt.off_color)
-            p.setRenderHint(QtGui.QPainter.RenderHint.Antialiasing, True)
-            p.setOpacity(0.8)
-        else:
-            p.setBrush(opt.off_color)
-            p.setOpacity(0.6)
-        return QtCore.QRect(
-            opt.margin,
-            self._vertical_offset(opt),
-            opt.switch_width,
-            opt.switch_height,
```

Comment:
this stuff really is draw stuff.  and should be overrideable in the `drawGroove` method.  (even though it does mean that the example gets slightly longer if someone wants to override `drawGroove`)